src/molecules/CheckboxGroup/CheckboxGroup.jsx

```diff
@@ -19,21 +19,24 @@ const CheckboxGroup = ({
 			newArr.push(sel);
 		}
 		updateSelected(newArr);
+		event.preventDefault();
 	};
 
 	return (
 		<div className="checkbox-group-wrapper">
-			{header && <h5>{header}</h5>}
-			{items.map((it, index) => (
-				<Checkbox
-					key={it.value}
-					index={index}
-					selected={selected.includes(it.value)}
-					value={it.value}
-					callback={setSelected}
-				>{it.label}
-				</Checkbox>
-			))}
+			<fieldset>
+				{header && <legend>{header}</legend>}
```

Comment:
I suggest wrapping the text in a `<Title size={5}>` inside of the legend. If so we can also remove the styling of the legend element in the css file.